pkg/controller/utils/image.go

```diff
@@ -1,23 +0,0 @@
-package utils
-
-import (
-	log "github.com/sirupsen/logrus"
-
-	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1alpha1"
-	"github.com/openshift/hive/pkg/controller/images"
-)
-
-// GetHiveImage looks for a Hive image to use in clusterdeployment jobs in the following order:
-// 1 - specified in the cluster deployment spec.images.hiveImage
-// 2 - referenced in the cluster deployment spec.imageSet
-// 3 - specified via environment variable to the hive controller
-// 4 - fallback default hardcoded image reference
-func GetHiveImage(cd *hivev1.ClusterDeployment, imageSet *hivev1.ClusterImageSet, cdLog log.FieldLogger) string {
-	if cd.Spec.Images.HiveImage != "" {
-		return cd.Spec.Images.HiveImage
-	}
-	if imageSet != nil && imageSet.Spec.HiveImage != nil {
-		return *imageSet.Spec.HiveImage
-	}
-	return images.GetHiveImage(cdLog)
-}
```

Comment:
This is getting added in https://github.com/openshift/hive/pull/482/commits/9279579d1387e1da2a45b511090e8c7c2407f519. Not sure why it is getting removed from here.

Reply:
The idea is that each commit is discrete and leaves the product in a buildable and usable state. In https://github.com/openshift/hive/commit/9279579d1387e1da2a45b511090e8c7c2407f519, the function was supposed to have been moved from ReconcileClusterDeployment to controllerutils. I see that in my restructuring of the commits, I did not remove it from ReconcileClusterDeployment. I will fix that commit to remove it.